centreon/src/Core/HostCategory/Infrastructure/Repository/DbReadHostCategoryRepository.php

```diff
@@ -508,7 +508,22 @@ private function retrieveHostCategories(
             'name' => 'hc.hc_name',
             'alias' => 'hc.hc_alias',
             'is_activated' => 'hc.hc_activate',
+            'host_group_id' => 'hg.hg_id',
         ]);
+        if (mb_strpos($requestParameters?->getSearchAsString() ?? '' , 'host_group_id')) {
+            $concatenator->appendJoins(
+                <<<'SQL'
+                        LEFT JOIN  hostcategories_relation hcr
+                            ON hc.hc_id = hcr.hostcategories_hc_id
+                        LEFT JOIN host h
+                            ON hcr.host_host_id = h.host_id
+                        LEFT JOIN hostgroup_relation hgr
+                            ON h.host_id = hgr.host_host_id
+                        LEFT JOIN hostgroup hg
+                            ON hgr.hostgroup_hg_id = hg.hg_id
```

Comment:
⚠️ If you are in an on-prem platform with non-admin user, this request is too open, 
you have to filter hostgroups, host and hostcategories on the user's ACLs

I talked about it with @dmyios, you can see with him for more detailed explanations